apps/emqx_machine/src/emqx_machine_schema.erl

```diff
@@ -282,6 +282,21 @@ fields("node") ->
        sc(ref("cluster_call"),
           #{}
          )}
+    , {"etc_dir",
+       sc(string(),
+          #{converter => fun(EtcDir) ->
+             case filename:absname(EtcDir) =:= EtcDir of
+                true -> binary_to_list(EtcDir);
+                false -> 
+                   AbsPath = filename:join([code:lib_dir(), "..", "etc"]),
+                   case filelib:is_dir(AbsPath) of
```

Comment:
please create a `validator` for this field to perform the validation.
so that line 289 is also checked with `filelib:is_dir/1`